### PR TITLE
Partially fix warn-stealers

### PR DESCRIPTION
**Pull request by wolfboyft**

### CODE REVIEW

**Comment by wolfboyft** [comment]
This failed me *again* during play, but it must be closer now...
At least, the changes I had locally did. Which were retyped in the web UI.

**Comment by wolfboyft** [comment]
The local changes and the changes in git are identical...

**Comment by wolfboyft** [comment]
It is possible it failed because I had messed up my local changes????

**Comment by myk002** [comment]
Changelog entry plz (is this a "Fixes" item or a "Misc Improvements" item?)

**Comment by wolfboyft** [comment]
I wonder if there is some way I can set up a reminder for changelog entries before make PRs... perhaps i'm just gonna have to forget a few more times 'til I start remembering.

File: changelog.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ that repo.
 ## Fixes
 - `gui/gm-unit`: fixed behavior of ``+`` and ``-`` again to adjust skill values instead of populating the search field
 - `gui/create-item`: restrict materials to those normally allowed by the game by default, introduce new ``--unrestricted`` option for full freedom in choosing materials
-- `warn-stealers`: fixed script not working because of an incorrect event name
+- `warn-stealers`: fixed script not working because of an incorrect event name, storing unit pointers instead of IDs, and it not announcing creatures that spawn in revealed
```
Comment (myk002) [inline changelog.txt:21]: There can be multiple lines for the same tool if the changes are different. I think this meets that criteria. How about a new line that says:
```
- `warn-stealers`: properly announce thieving creatures that spawn already revealed
```

**Comment by myk002** [comment]
> I wonder if there is some way I can set up a reminder for changelog entries before make PRs...

It's tough because not every PR deserves a changelog entry. E.g. a fix/rewrite for an earlier PR in the same release, a light editing pass to existing documentation, or a non-user-visible change to the build process.

On the other hand, it would be nice if new PRs were created with a note about changelog entries in the description field...

**Comment by wolfboyft** [comment]
Let me know if this changelog file is OK

**Comment by wolfboyft** [comment]
No entry for my most recent changes since they come under the unit->unitId changes

**Comment by wolfboyft** [comment]
WIP because of the issues shown in the discord

**Comment by wolfboyft** [comment]
Absolutely no idea how to continue with this

**Review by myk002 — APPROVED** [review]
I think these changes look good for what they do. I'd be happy merging this as is and if you can reproduce what appears to be the edge case (which I do not think is caused by the changes in this PR), it can be fixed in another PR. Does that sound ok?

**Comment by wolfboyft** [comment]
Sure thing, fixing this is in my super-long DF todo list so I'll get to it eventually

**Pull request merged**